block-producer/src/config.rs

```diff
@@ -15,6 +15,8 @@ pub struct BlockProducerConfig {
 
     /// Store gRPC endpoint in the format `http://<host>[:<port>]`.
     pub store_url: String,
+
+    pub verify_tx_proofs: bool,
```

Comment:
Let's add a doc comment here. I would also add a brief description of the motivation for this setting.